winpr/include/winpr/error.h

```diff
@@ -158,8 +158,8 @@ static INLINE HRESULT HRESULT_FROM_WIN32(unsigned long x)
 
 #define HRESULT_SEVERITY(hr) (((hr) >> 31) & 0x1)
 
-#define SUCCEEDED(hr) (((hr)) >= 0)
```

Comment:
sorry, this is not really good.
the define should check if a `HRESULT`, if you supply something else then you´re at fault.

Just change the variable you want to check to a `HRESULT` and do not touch the header.

Reply:
ok, sorry for the noise, looks like `MSDN` also proposes this (error hiding) way of defining `FAILED` :/

Reply:
ok. in reality I 1:1 copied that over from https://learn.microsoft.com/en-us/windows/win32/api/winerror/nf-winerror-succeeded

now I casted to HRESULT, hope this is how you meant.
As you said, the statusCode is defined as `statusCode (4 bytes): An unsigned integer representing errors that are detected by the RDG server`

Don't know if other protocol definitions are like this one, but MS-TSGU feels like it was written after they implemented this protocol and a manager forced them to document it.